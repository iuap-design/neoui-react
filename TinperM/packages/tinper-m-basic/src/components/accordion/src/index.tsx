import './Accordion.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import Accordion, { AccordionPanel } from './Accordion'

export type { AccordionProps, AccordionPanelProps } from './iAccordion'

export default attachPropertiesToComponent(Accordion, { Panel: AccordionPanel })
