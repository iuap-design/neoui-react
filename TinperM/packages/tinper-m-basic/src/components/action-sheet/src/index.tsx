import './ActionSheet.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import { showActionSheet } from './ActionSheet'
import ActionSheet from './ActionSheet'
export type { ActionSheetShowHandler, } from './ActionSheet'

export type {
  Action,
  ActionSheetProps,
} from './iActionSheet'

export default attachPropertiesToComponent(ActionSheet, { show: showActionSheet, })
