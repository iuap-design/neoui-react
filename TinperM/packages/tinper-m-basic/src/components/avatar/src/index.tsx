import './Avatar.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import Avatar from './Avatar'
import AvatarGroup from './AvatarGroup'

export type { AvatarProps } from './iAvatar'

export default attachPropertiesToComponent(Avatar, { Group: AvatarGroup, })
