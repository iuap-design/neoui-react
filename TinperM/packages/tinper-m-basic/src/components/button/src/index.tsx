import './Button.less'
import Button from './Button'
export default Button
