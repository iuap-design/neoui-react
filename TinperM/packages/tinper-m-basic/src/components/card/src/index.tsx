import './Card.less'
import Card from './Card'
export type { CardProps } from './iCard'

export default Card
