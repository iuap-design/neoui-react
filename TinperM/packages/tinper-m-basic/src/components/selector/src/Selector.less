@import  (reference)  '../../../theme/index.less';

.@{selector-prefix-cls} {
  --border-radius: ~"var(--@{selector-prefix-cls}-border-radius)";
  --color: ~"var(--@{selector-prefix-cls}-color-bg)";
  --checked-color: ~"var(--@{selector-prefix-cls}-color-bg-selected)";
  --checked-icon-color: ~"var(--@{selector-prefix-cls}-color-icon-selected)";
  --text-color: ~"var(--@{selector-prefix-cls}-color-text)";
  --checked-text-color: ~"var(--@{selector-prefix-cls}-color-text-selected)";
  --font-size: ~"var(--@{selector-prefix-cls}-font-size)";
  --item-height: ~"var(--@{selector-prefix-cls}-size-height)";
  --gap: ~"var(--@{selector-prefix-cls}-spacing-padding)";


  // --checked-color: var(--mui-color-wathet);
  --border: none;
  --checked-border: none;
  --padding: 0 0.16rem;
  ---gap: var(--gap);
  ---gap-horizontal: var(--gap-horizontal, 0.22rem);
  ---gap-vertical: var(--gap-vertical, 0.22rem);
  --flex: auto;
  --item-width: 1.44rem;

  overflow: hidden;
  font-size: var(--font-size);
  line-height: 1.4;

  &-with-no-badge {
    --checked-border: solid var(--mui-color-primary) 0.5px;
    --padding: 0.07rem 0.16rem;
    --border-radius: 0.08rem;

    &.@{selector-prefix-cls}-item {
      height: 100%;
    }
  }

  .mui-space.mui-space {
    --gap: 0.22rem;
    overflow-x: scroll;
    scrollbar-width: none;
  }


  .mui-space-item {
    flex: var(--flex);
    width: var(--item-width);


  }

  .mui-grid {
    --gap: var(---gap);
    --gap-horizontal: var(---gap-horizontal);
    --gap-vertical: var(---gap-vertical);
  }

  &-item {
    width: 100%;
    height: var(--item-height);
    padding: var(--padding);
    display: flex;
    flex-direction: column;

    align-items: center;
    justify-content: center;
    position: relative;
    background-color: var(--color);
    border: var(--border);
    border-radius: var(--border-radius);
    color: var(--text-color);
    opacity: 1;
    cursor: pointer;
    text-align: center;
    overflow: hidden;
    box-sizing: border-box;
    // vertical-align: top;
    &-label {
      overflow: hidden;
      word-break: break-all;
      text-overflow: ellipsis;
      display: -webkit-box;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
    }

    &-description {
      font-size: var(--mui-font-size-main);
    }

    &-active,
    &-multiple-active {
      color: var(--checked-text-color);
      background-color: var(--checked-color);
      border: var(--checked-border);
    }

    .@{selector-prefix-cls}-check-mark-wrapper {
      position: absolute;
      right: 0;
      bottom: 0;
      transform: translate(50%, 50%);
      width: 0.64rem;
      height: 0.64rem;
      background: ~"var(--@{selector-prefix-cls}-color-bg-mark-group-selector-selected)";
      border-radius: 50%;

      > svg {
        position: absolute;
        left: 25%;
        top: 25%;
        transform: translate(-25%, -25%);

        height: 0.12rem;
        width: 0.16rem;
      }
    }

    &-disabled {
      cursor: not-allowed;
      opacity: 0.5;
    }
  }
}
