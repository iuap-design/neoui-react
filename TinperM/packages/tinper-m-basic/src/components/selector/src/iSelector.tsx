import { FieldNamesType } from '@hooks/UseFieldNames';
import { NativeProps } from '@utils/NativeProps';
import type { ReactNode } from 'react';
import { BaseOptionType } from '@hooks/index';
import { GridProps } from '@components/grid/src';

export type SelectorProps<V> = {
  options: SelectorOption<V>[]
  columns?: GridProps['columns']
  multiple?: boolean
  disabled?: boolean
  defaultValue?: V[]
  value?: V[]
  onChange?: (v: V[], extend: { items: SelectorOption<V>[] }) => void
  showCheckMark?: boolean
  fieldNames?: FieldNamesType
  clsPrefix?: string
  widthAdjustment?: 'fixed' | 'auto' | 'equal'
  fieldid?: string
} & NativeProps<
  | '--color'
  | '--checked-color'
  | '--text-color'
  | '--checked-text-color'
  | '--border'
  | '--checked-border'
  | '--border-radius'
  | '--padding'
  | '--gap'
  | '--gap-vertical'
  | '--gap-horizontal'
  | '--item-width'
  | '--flex'
>


export type SelectorValue = string | number

export type SelectorOption<V> = {
  label?: ReactNode
  description?: ReactNode
  value?: V
  disabled?: boolean
} & BaseOptionType


type SelectorPropsCoverage = {
  options: any
  columns?: GridProps['columns']
  multiple?: boolean
  disabled?: boolean
  defaultValue?: any
  value?: any
  onChange?: any
    showCheckMark ?: boolean
  fieldNames?: FieldNamesType
  clsPrefix?: string
  widthAdjustment?: 'fixed' | 'auto' | 'equal'
  fieldid?: string
}
