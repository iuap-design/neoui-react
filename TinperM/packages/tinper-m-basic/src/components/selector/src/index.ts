import './Selector.less'
import { Selector } from './Selector'

export * from './iSelector'

export default Selector
