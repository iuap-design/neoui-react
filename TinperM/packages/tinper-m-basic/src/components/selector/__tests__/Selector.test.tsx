/** Selector.tsx */
import React from 'react'


import {render, screen} from "@testing-library/react";
import {Selector} from "@tinper/m";
import {fieldNamesOptions, options} from "@components/selector/demos/options";
import {SelectorProps, type SelectorValue} from "@components/selector/src";
import userEvent from "@testing-library/user-event";

describe('Selector Component ', () => {
  const renderSelector = <T extends SelectorValue>(props: Partial<SelectorProps<T>> = {}) => {
    const onChange = jest.fn();
    const {container} = render(<Selector options={options} onChange={onChange} {...props} />);
    const user = userEvent.setup()
    return {
      getSelectedOptions: () => screen.getAllByRole('option', {selected: true}),
      getOptionByLabel: (label: string) =>  screen.getByRole('option', {name: label}),
      getOptions: () => screen.getAllByRole('option'),
      user,
      selectOption: (index: number) => user.click(screen.getByText(options[index].label)),
      notSelected: () => screen.queryAllByRole('option', {selected: true}).length === 0,
      selector: container.firstElementChild,
      onChange,
      container
    }
  }
  it('should single selection by default <test prop:: defaultValue> <test prop:: onChange>', async () => {
    const {getSelectedOptions, selectOption, notSelected, onChange} = renderSelector({
      defaultValue: [options[0].value]
    })

    expect(getSelectedOptions()).toHaveLength(1)


    await selectOption(1)

    expect(getSelectedOptions()).toHaveLength(1)
    expect(onChange.mock.lastCall[0]).toEqual([options[1].value])
    expect(onChange.mock.lastCall[1].items).toEqual([options[1]])

    await selectOption(1)

    expect(notSelected()).toBe(true)
    expect(onChange.mock.lastCall[0]).toEqual([])
    expect(onChange.mock.lastCall[1].items).toEqual([])

  });

  it('should be controlled <test prop:: value> <test prop:: options>', async () => {
    const {getOptionByLabel, selectOption, notSelected, onChange} = renderSelector({
      value: [options[0].value]
    })

    await selectOption(1)

    expect(getOptionByLabel(options[0].label)).toHaveClass(/active/)



  });

  it('should multiple selection <test prop:: multiple>', async () => {
    const {getSelectedOptions, selectOption, onChange} = renderSelector({
      defaultValue: [options[0].value, options[1].value],
      multiple: true
    })

    await selectOption(2)

    expect(getSelectedOptions()).toHaveLength(3)

    await selectOption(1)

    expect(getSelectedOptions()).toHaveLength(2)

    await selectOption(1)

    expect(getSelectedOptions()).toHaveLength(3)


  });

  it('should be fixed width <test prop:: widthAdjustment>', () => {
    const {selector} = renderSelector({widthAdjustment: 'fixed', style: {'--item-width': '200px'}});
    expect(selector).toMatchSnapshot()
  });

  it('width should be determined by the content <test prop:: widthAdjustment>', () => {
    const {selector} = renderSelector({widthAdjustment: 'auto'});
    expect(selector).toMatchSnapshot()
  });

  it('should be 2 columns <test prop:: columns>', () => {
    const {selector} = renderSelector({columns: 2});
    expect(selector).toMatchSnapshot()
  });

  it('should be disabled <test prop:: disabled>', async () => {
    const {getOptions, selectOption} = renderSelector({disabled: true});
    const options = getOptions()

    options.forEach(option => {
      expect(option).toHaveClass(/disabled/)
    })

    await selectOption(2)

    expect(options[2]).toHaveClass(/disabled/)
  });

  it('the FieldName should be customized <test prop:: fieldNames>', () => {
    const { getOptionByLabel } = renderSelector({
      options: fieldNamesOptions,
      defaultValue: [fieldNamesOptions[1].valueT],
      fieldNames: {
        label: 'labelT',
        value: 'valueT',
        disabled: 'disabledT',
      }
    });

    fieldNamesOptions.forEach(option => {
      expect(screen.getByText(option.labelT)).toBeInTheDocument()
    })

    expect(getOptionByLabel(fieldNamesOptions[1].labelT)).toHaveClass(/active/)

    expect(getOptionByLabel(fieldNamesOptions[2].labelT)).toHaveClass(/disabled/)

  });

  it('should render description', () => {
    renderSelector({
      options: [
        {
          label: '选项1',
          description: '描述信息',
          value: '1',
        }]
    })

    expect(screen.getByText('描述信息')).toBeInTheDocument()
  });

  it('should not render checkMark <test prop:: showCheckMark>', () => {
     const { getSelectedOptions } = renderSelector({showCheckMark: false, defaultValue: [options[0].value]});

    expect(getSelectedOptions()[0]).toMatchSnapshot()
  });

  it('should contain fieldid <test prop:: fieldid>', () => {
    const fieldid = 'fieldid-selector'
    const { selector } = renderSelector({fieldid});

    expect(selector).toHaveAttribute('fieldid', fieldid)

  });

  it('should contain clsPrefix <test prop:: clsPrefix>', () => {

    const { container } = renderSelector({ clsPrefix: 'clsPrefix' });

    expect(container.firstElementChild).toHaveClass(/clsPrefix/i)

  });


})
