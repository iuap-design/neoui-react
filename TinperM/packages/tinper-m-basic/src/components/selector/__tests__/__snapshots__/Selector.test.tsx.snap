// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Component  should be 2 columns <test prop:: columns> 1`] = `
<div
  class="mui-selector"
  role="listbox"
  style="--flex: auto; --item-width: 1.44rem;"
>
  <div
    class="mui-grid"
    style="--columns: 2;"
  >
    <div
      aria-selected="false"
      class="mui-selector-item"
      fieldid="undefined-1"
      role="option"
    >
      <div
        class="mui-selector-item-label"
      >
         
        选项1
      </div>
    </div>
    <div
      aria-selected="false"
      class="mui-selector-item"
      fieldid="undefined-2"
      role="option"
    >
      <div
        class="mui-selector-item-label"
      >
         
        选项2
      </div>
    </div>
    <div
      aria-selected="false"
      class="mui-selector-item"
      fieldid="undefined-3"
      role="option"
    >
      <div
        class="mui-selector-item-label"
      >
         
        选项3
      </div>
    </div>
  </div>
</div>
`;

exports[`Selector Component  should be fixed width <test prop:: widthAdjustment> 1`] = `
<div
  class="mui-selector"
  role="listbox"
  style="--flex: none; --item-width: 200px;"
>
  <div
    class="mui-space mui-space-block mui-space-horizontal"
    style="--gap: 0.08rem;"
  >
    <div
      class="mui-space-item"
    >
      <div
        aria-selected="false"
        class="mui-selector-item"
        fieldid="undefined-1"
        role="option"
      >
        <div
          class="mui-selector-item-label"
        >
           
          选项1
        </div>
      </div>
    </div>
    <div
      class="mui-space-item"
    >
      <div
        aria-selected="false"
        class="mui-selector-item"
        fieldid="undefined-2"
        role="option"
      >
        <div
          class="mui-selector-item-label"
        >
           
          选项2
        </div>
      </div>
    </div>
    <div
      class="mui-space-item"
    >
      <div
        aria-selected="false"
        class="mui-selector-item"
        fieldid="undefined-3"
        role="option"
      >
        <div
          class="mui-selector-item-label"
        >
           
          选项3
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Selector Component  should not render checkMark <test prop:: showCheckMark> 1`] = `
<div
  aria-selected="true"
  class="mui-selector-item mui-selector-item-active mui-selector-with-no-badge"
  fieldid="undefined-1"
  role="option"
>
  <div
    class="mui-selector-item-label"
  >
     
    选项1
  </div>
</div>
`;

exports[`Selector Component  width should be determined by the content <test prop:: widthAdjustment> 1`] = `
<div
  class="mui-selector"
  role="listbox"
  style="--flex: none; --item-width: auto;"
>
  <div
    class="mui-space mui-space-block mui-space-horizontal"
    style="--gap: 0.08rem;"
  >
    <div
      class="mui-space-item"
    >
      <div
        aria-selected="false"
        class="mui-selector-item"
        fieldid="undefined-1"
        role="option"
      >
        <div
          class="mui-selector-item-label"
        >
           
          选项1
        </div>
      </div>
    </div>
    <div
      class="mui-space-item"
    >
      <div
        aria-selected="false"
        class="mui-selector-item"
        fieldid="undefined-2"
        role="option"
      >
        <div
          class="mui-selector-item-label"
        >
           
          选项2
        </div>
      </div>
    </div>
    <div
      class="mui-space-item"
    >
      <div
        aria-selected="false"
        class="mui-selector-item"
        fieldid="undefined-3"
        role="option"
      >
        <div
          class="mui-selector-item-label"
        >
           
          选项3
        </div>
      </div>
    </div>
  </div>
</div>
`;
